Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -280,6 +280,24 @@ LayoutUnit GridTrackSizingAlgorithm::initialGrowthLimit(const GridTrackSize& tra
     return infinity;
 }
 
+bool GridTrackSizingAlgorithm::isUnderMaxContentConstraint() const
+{
+    bool noAvailableSpace = !availableSpace(m_direction).has_value();
+    // Also check if the grid has an explicit max-content size
+    if (m_direction == GridTrackSizingDirection::ForColumns) {
+        const auto& gridStyle = m_renderGrid->style();
+        // Check if width is explicitly set to max-content
+        if (gridStyle.width().isMaxContent())
```

Comment:
This is still wrong, it's not a matter of the computed style, it's a matter of whether the grid is being sized under a min/max-constraint, or the size of the grid container is known and the grid is being laid out into there.

Blink tracks this with

```cpp
enum class SizingConstraint { kLayout, kMaxContent, kMinContent };
```